private/README.md

```diff
@@ -0,0 +1,2 @@
+## Smithy Go Private packages ##
+`private` is a collection of packages used internally by Smithy Go, and is subject to have breaking changes. This package is not `internal` so that if you really need to use its functionality, and understand breaking changes will be made, you are able to.
```

Comment:
I think just the first sentence is fine.

Reply:
> I think just the first sentence is fine.

Ya, I would either drop the rest or if you feel to need to say more I'd change it to something like `This package is not `internal` because it is an implementation detail of generated code.`

Reply:
@lucix-aws sure, i can do that. i was just copying how both the v1 and v2 repository use a root level internal directory. but i dont have any objections with the way you suggested above.

@aajtodd sure, removed the requested lines.